### PR TITLE
Db sync

### LINKED ISSUE #10: DB_SYNC

### PR DESCRIPTION
**Pull request by RobertoAldea**

### CODE REVIEW

File: .github/workflows/deploy_db_sync.yml
Comment (mariuscrsn) [inline .github/workflows/deploy_db_sync.yml:1]: Use the same workflow for cyclers datatypes, cu_manager and db_sync

File: code/db_sync/LICENSE.txt
Comment (mariuscrsn) [inline code/db_sync/LICENSE.txt:1]: Use the same LICENSE for all pacakges

Comment (mariuscrsn) [reply code/db_sync/LICENSE.txt:1]: It should be placed on the repository root folder and link it to the pacakges

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        self.fachade.push_gen_meas()
+        self.fachade.commit()
+        log.debug("Commit and push of gen and exp done")
+        self.fachade.push_ext_meas()
+        self.fachade.push_alarms()
+        self.fachade.push_status()
+        self.fachade.commit()
+        log.debug("Commit and push ext, alarms and status done")
+        self.fachade.delete_pushed_data()
+        self.fachade.update_last_connection()
```
Comment (mariuscrsn) [inline code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py]: Missing a last commit?

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            if (meas.ExpID not in self.__last_exp_status or (meas.ExpID in self.__last_exp_status
+                and meas.Status != self.__last_exp_status[meas.ExpID])):
+                self.__last_exp_status[meas.ExpID] = meas.Status
+                log.info(f"Experiment {meas.ExpID} status changed to {meas.Status}")
+                transform_experiment_db(source= meas, target= meas_add)
+                self.__push_exps.append(meas)
+                self.__master_db.session.merge(meas_add)
+
+
+    def update_last_connection(self) -> None:
```
Comment (mariuscrsn) [inline code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py]: Remove this functions, now it is implemented though MQTT

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        Args:
+            - None
+        Returns:
+            - None
+        Raises:
+            - None
+        '''
+        log.info("Commiting changes...")
+        try:
+            # pass
```
Comment (mariuscrsn) [inline code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py]: Clean code, remove it

File: code/db_sync/src/wattrex_battery_cycler_db_sync/context.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+#######################          PROJECT IMPORTS         #######################
+from system_config_tool import sys_conf_update_config_params
+
+#######################          MODULE IMPORTS          #######################
+
+######################             CONSTANTS              ######################
+# For further information check out README.md
+DEFAULT_CRED_FILEPATH : str = 'devops/.cred.yaml' # Max number of allowed message per chan
+DEFAULT_SYNC_NODE_NAME: str = 'SYNC'
```
Comment (mariuscrsn) [inline code/db_sync/src/wattrex_battery_cycler_db_sync/context.py:27]: Add comp_unit and cycle_period from db_sync_node

File: .github/workflows/deploy_db_sync.yml
Comment (mariuscrsn) [inline .github/workflows/deploy_db_sync.yml:1]: Missing an run_db_sync example on devops/db_sync

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py (hunk 1 above)
Comment (RobertoAldea) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py]: If you refer to commit in cache is inside delete_pushed_data

File: .github/workflows/deploy_db_sync.yml
Comment (javibu13) [reply .github/workflows/deploy_db_sync.yml:1]: image

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py (hunk 3 above)
Comment (RobertoAldea) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py]: Done

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py (hunk 2 above)
Comment (RobertoAldea) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py]: OK

File: code/db_sync/src/wattrex_battery_cycler_db_sync/context.py (hunk 4 above)
Comment (RobertoAldea) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/context.py:27]: OK

File: .github/workflows/deploy_db_sync.yml
Comment (javibu13) [reply .github/workflows/deploy_db_sync.yml:1]: Done.

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py (hunk 2 above)
Comment (mariuscrsn) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_fachade.py]: Remove it

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
```
Comment (mariuscrsn) [inline code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py]: Remove function `self.fachade.update_last_connection()`. It is performed on MQTT node

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py (hunk 5 above)
Comment (RobertoAldea) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py]: Done

Comment (RobertoAldea) [reply code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py]: I have done some modifications when pushing data. Also move the try except to follow better the status machine of doc

File: code/db_sync/LICENSE.txt
Comment (javibu13) [reply code/db_sync/LICENSE.txt:1]: Done in db_sync package (.toml) and added in system action build-package the copy of the LICENSE.txt

Comment (mariuscrsn) [reply code/db_sync/LICENSE.txt:1]: Good idea!

File: code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py [hunk 6]
```diff
@@ -77,12 +77,14 @@ def process_iteration(self) -> None:
...
+        try:
+            self.fachade.commit()
+            log.debug("Commit and push of gen and exp done")
+            self.fachade.push_ext_meas()
+            self.fachade.push_alarms()
+            self.fachade.push_status()
+            self.fachade.commit()
+            log.debug("Commit and push ext, alarms and status done")
+            self.fachade.delete_pushed_data()
+        except Exception as err:
```
Comment (mariuscrsn) [inline code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py:89]: Do you really execute rollback if something fails? I think you should catch the exception and do the rollback in the  commit and delete_pushed_data functions

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Pull request merged**